.github/workflows/self_hosted_build_push.yaml

```diff
@@ -0,0 +1,155 @@
+      - name: Login to ECR
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ inputs.AWS_ECR_REGISTRY }}
+          username: ${{ secrets.ECR_AWS_ACCESS_KEY_ID }}
+          password: ${{ secrets.ECR_AWS_SECRET_ACCESS_KEY }}
+        env:
+          AWS_REGION: ${{ inputs.AWS_REGION }}
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          driver: kubernetes
+          driver-opts: namespace=self-hosted-runners,"nodeselector=kubernetes.io/arch=amd64,node.kubernetes.io/instance-type=t3.2xlarge",requests.cpu=6,requests.memory=25G,"tolerations=key=role,value=armrunners,effect=NoSchedule"
+          platforms: linux/amd64
+          append: |
+            - platforms: linux/arm64
+              name: k8s-arm${{ inputs.OVERRIDE_BUILDER_NAME }}
+              driver-opts:
+              - namespace=self-hosted-runners,"nodeselector=kubernetes.io/arch=arm64,node.kubernetes.io/instance-type=c7g.2xlarge",requests.cpu=6,requests.memory=10G,"tolerations=key=role,value=armrunners,effect=NoSchedule"
+
+      - name: Set Variables
+        shell: bash
+        run: |
+          echo "OVERRIDE_REPO_NAME = ${{ inputs.OVERRIDE_REPO_NAME }}"
+          echo "OVERRIDE_TAG_NAME = ${{ inputs.OVERRIDE_TAG_NAME }}"
+
+          if [[ -z "${{ inputs.OVERRIDE_TAG_NAME }}" ]]
+          then
+            echo "No OVERRIDE_TAG_NAME input provided, defaulting to current branch/tag name..."
+            echo "IMAGE_TAG=$(echo ${GITHUB_REF#refs/*/} | tr / _)"
+            echo "IMAGE_TAG=$(echo ${GITHUB_REF#refs/*/} | tr / _)" >> $GITHUB_ENV
+          else
+            echo "OVERRIDE_TAG_NAME provided, using it for IMAGE_TAG..."
+            echo "IMAGE_TAG=${{ inputs.OVERRIDE_TAG_NAME }}"
+            echo "IMAGE_TAG=${{ inputs.OVERRIDE_TAG_NAME }}" >> $GITHUB_ENV
+          fi
+
+          if [[ -z "${{ inputs.OVERRIDE_REPO_NAME }}" ]]
+          then
+            echo "No OVERRIDE_REPO_NAME input provided, defaulting to repo name..."
+            echo "REPO_NAME=$(echo $GITHUB_REPOSITORY | awk -F / '{print $2}')"
+            echo "REPO_NAME=$(echo $GITHUB_REPOSITORY | awk -F / '{print $2}')" >> $GITHUB_ENV
+          else
+            echo "OVERRIDE_REPO_NAME provided, using it for REPO_NAME..."
+            echo "REPO_NAME=${{ inputs.OVERRIDE_REPO_NAME }}"
+            echo "REPO_NAME=${{ inputs.OVERRIDE_REPO_NAME }}" >> $GITHUB_ENV
+          fi
+
+      - name: Extract metadata
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: |
+            quay.io/cdis/${{ env.REPO_NAME }}:${{ env.IMAGE_TAG }}
+            ${{ inputs.AWS_ECR_REGISTRY }}/gen3/${{ env.REPO_NAME }}:${{ env.IMAGE_TAG }}
+
+      - name: Build and push ECR
+        if: ${{ !inputs.USE_QUAY_ONLY }}
+        uses: docker/build-push-action@v3
```

Comment:
Bump docker/build-push-action@v3  to **docker/build-push-action@v4**

Reply:
Done